src/agentlab/analyze/agent_xray.py

```diff
@@ -627,11 +624,16 @@ def submit_action(input_text):
     global info
     agent_info = info.exp_result.steps_info[info.step].agent_info
     chat_messages = deepcopy(agent_info.get("chat_messages", ["No Chat Messages"])[:2])
-    assert isinstance(chat_messages[1], BaseMessage), "Messages should be langchain messages"
-
-    chat = ChatOpenAI(name="gpt-4o-mini")
-    chat_messages[1].content = input_text
-    result_text = chat(chat_messages).content
+    assert isinstance(chat_messages[1], dict), "Messages should be a dict"
```

Comment:
could we make backward compatible code here as well with a `# TODO remove once
langchain is deprecated`